src/packages/database/model/initialize-class.js

```diff
@@ -269,7 +269,7 @@ export default async function initializeClass<T: Class<Model>>({
 
       Object.defineProperties(relationship, {
         model: {
-          value: store.modelFor(relatedModel || singularize(relatedName)),
+          value: store.modelFor(relatedModel || relatedName),
```

Comment:
No need to `singularize` before calls to `store.modelFor` as normalization now takes place before the key is retrieved from the `Map` stored in `Database#models`.